### ISSUE TITLE
Bug: controller (dualshock 4) doesn't work when launching P5R via reloaded II

### ISSUE THREAD

**Issue by shivvver** [Reloaded-Project/Reloaded-II#185]
**Describe the Bug**
My dualshock 4 controller doesn't work when launching the persona 5 royal through reloaded II

**To Reproduce**
launch persona 5 royal through reloaded-II

**Expected Behavior**
I expected my controller to work

**Screenshots**
![image](https://user-images.githubusercontent.com/123616000/214759366-8fefe254-10d3-43fe-8f88-0ffeed93e9c3.png)
ugly ass cursor

**Additional Context**
it worked yesterday after relaunching many times, no luck after launching it even more times today.
also, when it did work, i would get the notification in the bottom right corner for the steam community and then the controller, but now i don't get the notification for either

**Comment by Sewer56** [comment]
I'm guessing your controller is being stolen by Steam and unavailable to applications launched outside of it.

Try `Edit Application -> Advanced Tools & Options -> Deploy ASI Loader`, and then launching game through Steam.

**Comment by shivvver** [comment]
now i'm getting "unknown hard error" when trying to launch

**Comment by Sewer56** [comment]
Can you take a screenshot? I've never heard of this kind of error before.

**Comment by shivvver** [comment]
![image](https://user-images.githubusercontent.com/123616000/215231685-3b19e935-c9e1-4da3-abd9-7aca06637a5f.png)

**Comment by shivvver** [comment]
it's just a black screen, and then the error pops up

**Comment by Sewer56** [comment]
I guess without ASI Loader (delete `dinput8.dll` in game folder) it still runs normally, correct?

**Comment by shivvver** [comment]
okay wait why does this work

**Comment by shivvver** [comment]
bruh

**Comment by shivvver** [comment]
well it worked for a minute... XD

**Comment by shivvver** [comment]
yeah, no luck trying to recreate that either

**Comment by Sewer56** [comment]
Well; this is awkward.

**Comment by shivvver** [comment]
yeah...

**Comment by shivvver** [comment]
i'm trying verifying the game files

**Comment by shivvver** [comment]
still nothing...

**Comment by Sewer56** [comment]
> I guess without ASI Loader (delete `dinput8.dll` in game folder) it still runs normally, correct?

Can you confirm if this is ASI Loader specific?

**Comment by shivvver** [comment]
the hard error only happens when ASI loader is in the files so... yeah probably, if that's what you mean

**Comment by Sewer56** [comment]
Yeah... pretty strange honestly.

**Comment by shivvver** [comment]
although steam did give me a notification that i could access the community and the controller was connected when it originally so maybe it has to do with how reloaded is interacting with my steam session?

**Comment by Sewer56** [comment]
What I've experienced in the past is Steam can open the controller in Exclusive Mode (or something of that sort) meaning that any program that isn't launched through Steam's launcher at all wouldn't have functional controller support (at all).

That's what probably is the case here. As an experiment, uncheck `PlayStation Controller Support` in Steam Big Picture; and consider installing DS4Windows (if you haven't already) to emulate a 360 controller and hide the original controller. See how that goes.

**Comment by shivvver** [comment]
So, get this:
I go into steam big picture mode
I see that "Playstation Controller Support" is unchecked, and I think to myself "Huh, I didn't do that."
I check Playstation Controller Support
Lo and behold, it worked.
What in the Sam f*ck.

**Comment by shivvver** [comment]
Sorry I had to bug you about this, it seems so stupid now.
Really though, thank you so much for your help, I still probably couldn't have done it without you.

**Comment by shivvver** [comment]
okay, this was only because steam gave it the desktop controls, but it works, so i'm happy

**Comment by shivvver** [comment]
okay i lied camera and movement dont work

**Comment by Sewer56** [comment]
I fell asleep but I have a DualShock 4 laying around, I'll be able to check myself soon.

**Comment by shivvver** [comment]
If yours works, I feel like if we could fix anything that could be causing the steam community tab notification to not show, it would fix the controller problem too.

**Comment by shivvver** [comment]
still no dice?

**Comment by Sewer56** [comment]
DualShock 4 seems to work for me under plain old Steam Input; regardless of whether ASI Loader is installed or not.

![explorer_5jBj5tldMb](https://user-images.githubusercontent.com/6697380/219220453-814bfa60-c7a7-4350-a731-daf13f46fef2.png)
![P5R_dKLH8HkUMi](https://user-images.githubusercontent.com/6697380/219220457-89425c48-61a1-4667-8a70-61db051efbf4.png)

**Comment by Novationo** [comment]
I recently encountered this problem as well. I've tried deploying ASI and changing controller configuration support in Steam to no prevail. 
This started **after I played P5R through Parsec**, playing with joycons or a controller connected to my phone works, but now controllers do not work when playing normally on my pc. Ive tried using both a Dualsense and a Xbox Series X controller and neither work now, and closing Parsec hasn't done anything. 
When trying to debug, P5R now crashes when launching through Steam, I've verified my game files and that did not fix it.

**Comment by EternisRequiem** [comment]
I also can't load P5R through Steam anymore with ASI (P5R will load without mods again if I delete dinput8.dll); loading from reloaded manually works, as does a steam game shortcut, however Steam Overlay (and therefore controllers) does not load/work with anything other than the vanilla game ...

**Comment by Novationo** [comment]
This is an update to my previous comment:
I reinstalled the game + all my mods, it worked until I installed [Battle Music Randomizer](https://gamebanana.com/mods/412327), but I was able to remedy it by disabling `imGui Menu` in the configure mod settings, then my controller worked again!

If you have this mod installed, try this, see if it works.
Also if you intend on verifying your game install or reinstalling, back up your savedata just to be safe, its located in `%appdata%\SEGA\P5R`.
